mcs/class/System.Drawing/Test/System.Drawing.Drawing2D/GraphicsPathIteratorTest.cs

```diff
@@ -73,85 +73,79 @@ public void NextSubpath_Null ()
 		}
 
 		[Test]
-		[ExpectedException (typeof (NullReferenceException))]
 		public void CopyData_NullPoints ()
 		{
 			using (GraphicsPath gp = new GraphicsPath ()) {
 				gp.AddLines (pts_2f);
 				using (GraphicsPathIterator gpi = new GraphicsPathIterator (gp)) {
 					PointF [] points = null;
 					byte [] types = new byte [1];
-					Assert.AreEqual (0, gpi.CopyData (ref points, ref types, 0, 1));
+					Assert.Throws<NullReferenceException> (() => Assert.AreEqual (0, gpi.CopyData (ref points, ref types, 0, 1)));
```

Comment:
Yeah, there are a couple of instances like this. Let me first get this PR to a point where all the tests pass (i.e. to make sure the Assert.Throws is at the correct location), I'll do a cleanup of these instances after that.